src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java

```diff
@@ -147,10 +148,22 @@
       + " mapping conflict or a field name containing illegal characters. Valid options are "
       + "'ignore', 'warn', and 'fail'.";
 
+  public static final String CONNECTION_SSL_CONFIG = "elastic.https";
```

Comment:
Would you mind adding some background/context on why this config is needed and protocol in url is not enough ? (basically some motivation at the initial commit message of this PR)

Reply:
The original thinking was that offering an explicit, documented option gives less mysterious reasons for misconfiguration. Another motivation is today we allow ambiguous protocols in the host (e.g. `localhost` instead of `http://localhost`). However I can be convinced that this is unnecessary and we can leave it out, and rely on the rule of "if any of the connection URLs starts with HTTPS, we will use HTTPS across the board"). 

This config does raise one ambiguity I don't consider: What if user specifies `https://localhost` and sets `elastic.https=false`? Maybe we should throw an error.

I can be convinced either way, but now i'm leaning slightly towards removing the option. Thoughts?

Reply:
I agree that it'd be better to specify it in one place, preferably in the URL protocol. Might be good to make the `elastic.https.*` configs dependent upon the URL config, such that they only show up when the protocol is `https`. Thoughts?

Reply:
Perhaps if the UI disables the fields and gives the user some documentation on how to use it, then I could get behind hiding the fields, but I think hiding them completely by default would be confusing. I don't know enough about our GUIs to make the call here what is possible.

If we are thinking "cloud-first" then we should probably ask explicitly a boolean question ("Should use SSL?" and have everything else be subfields. In this case I think the best config-file experience is if this config completely controls ssl decision, and use a validator/recommender. I can also add a log warning around the security code if they try to use a keystore but don't enable the boolean.

I'm fairly confident that parsing URLs is not the right move here after this discussion, and we should stick to boolean only. Thoughts?

Reply:
Thanks for the discussion, @kkonstantine , @rhauch , if you guys agree with the decision here then I can push what should be a final version

Reply:
Hmm, I'm not convinced the config is absolutely necessary. Checking the protocol is pretty common approach. And in the absence of protocol a sensible default may be used. 

We already do that elsewhere: 
https://github.com/confluentinc/kafka-connect-storage-cloud/blob/master/kafka-connect-s3/src/main/java/io/confluent/connect/s3/util/S3ProxyConfig.java#L62-L67

I think my suggestion would be to assume `https` by default (if protocol is absent) and `http` if explicitly present. WDYT?

Reply:
@rhauch @kkonstantine @cyrusv I will rely on your technical expertise. One thing I'd like to make a point here is what's best for users? Can we make TLS/SSL configuration consistent across connectors? It seems Replicator and S3 have their own way to handle it. How do other connectors support TLS/SSL?

Reply:
Replicator and MQTT proxy hav an enum option for "kafka.security.protocol" and you can pick PLAINTEXT, SSL, or SASL, and get the relevant subfields. S3 proxy does not have any subfields for passing certificates and passwords around, so not sure it's the best analogy here. If anything i would upgrade this to an enum for PLAINTEXT or SSL. not sure if sasl or other mechanisms might be used in the future. I can read more of our connectors to see if there is more to the pattern.

Reply:
After discussion w @kkonstantine, since Elastic is only HTTP(S) it is fine to go with the simpler pattern we use in s3 and determine protocol from the URL